File: src/main/java/org/example/kotlinWorkshop/java/lesson02/t5_null/t5_2/NullCoalesceOperator.java
package org.example.kotlinWorkshop.java.lesson02.t5_null.t5_2;

import org.jetbrains.annotations.NotNull;

public class NullCoalesceOperator {
    private class AnObject {
        private String name;

        public AnObject(String name) {
            this.name = name;
        }

        public String getName() {
            return name;
        }
    }

    private AnObject anObject;

    public NullCoalesceOperator(AnObject anObject) {
        this.anObject = anObject;
    }

    public AnObject getAnObject() {
        return anObject;
    }

    @NotNull
    public String getAnObjectNameOrDefault() {
        if(anObject == null || anObject.name == null) {
            return "default";
        }
        return anObject.name;
    }
}

File: src/main/kotlin/org/example/kotlinWorkshop/kotlin/lesson02/t5_null/t5_2/NullCoalesceOperator.kt
package org.example.kotlinWorkshop.kotlin.lesson02.t5_null.t5_2

class NullCoalesceOperator(private val anObject: AnObject?) {
    inner class AnObject(val name: String?)

    fun getAnObjectNameOrDefault(): String {
        return anObject?.name ?: "default"
    }
}